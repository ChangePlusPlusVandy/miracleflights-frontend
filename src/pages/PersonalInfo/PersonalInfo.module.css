.tabs {
  width: 100%;
  font-family: "Inter", serif;
}

.tabList {
  /*applies to the entire box of tabs */
  border-bottom: 1px solid gray;
  margin: 0 0 10px; /*top right/left bottom-applies to separate the text for each tab from the tablist */
  width: 100%;
  flex-direction: row;
  justify-content: center;
  display: flex;
}

.tab {
  /*applies to each individual tab */
  border: none;
  color: gray;
  background: transparent;
  margin: 10px 20px;
  font-size: 90%;
  padding: 6px 12px;
  cursor: pointer;
  transition: 200ms ease-in-out;
}

.tab:hover {
  /*applies to each individual tab */
  transform: translateY(-3px);
}

.tabSelected {
  border: none;
  border-bottom: 2px solid #cf2e2e;
  color: black;
  background: transparent;
  margin: 0 20px;
  font-size: 90%;
  padding: 6px 12px;
  cursor: pointer;
  font-weight: bold;
  transition: 200ms ease-in-out;
}

.tabPanel {
  display: flex;
  justify-content: center;
  font-weight: bold;
}
