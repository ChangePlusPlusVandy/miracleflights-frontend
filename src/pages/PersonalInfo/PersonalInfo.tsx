const PersonalInfo = () => {
  // Personal info tab
  console.log("PersonalInfo");

  return <div>PersonalInfo</div>;
};

export default PersonalInfo;
