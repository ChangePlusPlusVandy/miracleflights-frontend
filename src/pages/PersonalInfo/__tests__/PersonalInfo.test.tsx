import PersonalInfo from "../PersonalInfo";
import { render, fireEvent } from "@testing-library/react";

/**************************TABS TESTING SUITE*******************************/
describe("Tabs Tests", () => {
  /**************************FIRST TEST*******************************/
  test("Test tabs", () => {
    const tab = render(<PersonalInfo />);

    // Check that the tabs render
    expect(tab).toBeTruthy();

    // Get the 1st and 2nd tab buttons
    const tab1 = tab.getByText("One");
    const tab2 = tab.getByText("Two");
    const tab3 = tab.getByText("Three");
    const tab4 = tab.getByText("Four");
    const tab5 = tab.getByText("Five");

    // Check that the 1st tab displays correct info
    fireEvent.click(tab1);
    expect(tab.getByText("hello i am 1")).toBeTruthy();

    // Check that the 2nd tab displays the correct info
    fireEvent.click(tab2);
    expect(tab.getByText("hello i am 2")).toBeTruthy();

    // Check that the 1st tab displays correct info
    fireEvent.click(tab3);
    expect(tab.getByText("hello i am 3")).toBeTruthy();

    // Check that the 2nd tab displays the correct info
    fireEvent.click(tab4);
    expect(tab.getByText("hello i am 4")).toBeTruthy();
    // Check that the 1st tab displays correct info
    fireEvent.click(tab5);
    expect(tab.getByText("hello i am 5")).toBeTruthy();
  });

  test("Test snapshots", () => {
    const { container } = render(<PersonalInfo />);
    expect(container).toMatchSnapshot();
  });
});
