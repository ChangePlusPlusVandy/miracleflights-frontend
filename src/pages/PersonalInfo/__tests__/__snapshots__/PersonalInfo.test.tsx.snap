// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Tests Test snapshots 1`] = `
<div>
  <div
    class="tabs"
  >
    <div
      class="tabList"
    >
      <button
        class="tabSelected"
        type="button"
      >
        One
      </button>
      <button
        class="tab"
        type="button"
      >
        Two
      </button>
      <button
        class="tab"
        type="button"
      >
        Three
      </button>
      <button
        class="tab"
        type="button"
      >
        Four
      </button>
      <button
        class="tab"
        type="button"
      >
        Five
      </button>
    </div>
    <div
      class="tabPanel"
    >
      hello i am 1
    </div>
  </div>
</div>
`;
